src/switch/get-peer-info.js

```diff
@@ -0,0 +1,49 @@
+'use strict'
+
+const PeerId = require('peer-id')
+const PeerInfo = require('peer-info')
+const multiaddr = require('multiaddr')
+
+/**
+ * Helper method to check the data type of peer and convert it to PeerInfo
+ *
+ * @param {PeerInfo|Multiaddr|PeerId} peer
+ * @param {PeerBook} peerBook
+ * @throws {InvalidPeerType}
+ * @returns {PeerInfo}
+ */
+function getPeerInfo (peer, peerBook) {
```

Comment:
This will probably make sense in other contexts. So, after getting all other modules in here, we should also analyze all the utils and put them in a common util folder